kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -264,16 +279,23 @@ private static ColumnarBatch transformPhysicalRowId(
                 new ScalarExpression(
                     "ADD",
                     Arrays.asList(
-                        new Column(StructField.METADATA_ROW_INDEX_COLUMN_NAME),
+                        new Column(
+                            dataBatch
+                                .getSchema()
+                                .at(dataBatch.getSchema().indexOf(MetadataColumn.ROW_INDEX))
+                                .getName()),
```

Comment:
Can you store this in a variable `rowIndexMetadataColName` before? Will make this easier to parse